### DISCUSSION TITLE
Softconsole targeting MiV & debugging code in external RAM

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#166] [Bare metal embedded software] [answered]
Have a customer that is using the PolarFire and in particular the MiV.    His goal is to develop a bootloader and the sequence would be boot from TCM, that code validates and loads an image from NVM, (some flash) into external DDR and then execute from that DDR.  classic bootloader use case   here's the issue...

today they can load code to the TCM via SoftConsole, that code can access the external RAM/DDR no problem, so the plumbing is there to support this bootloader.    However the goal is to debug code via SoftConsole executing out of the external RAM/DDR.     But..  when he builds his image to reside in the external RAM he cannot load via SoftConsole and it makes sense, no one has setup the DDR at this point.    

The System Controller in this case has setup the TCM so how do we get the logic enabled to setup use of the external RAM/DDR via SoftConole?

Thx!
Dave

**Comment by hughbreslin** [comment, marked as answer]
Hey Dave,

this isn't one for this forum unfortunately, our [guidelines](https://github.com/orgs/polarfire-soc/discussions/125) state we can only deal with queries related to this GitHub organization. 

Just as something though, I think you want to follow the same methodology in the answer for [discussion 100](https://github.com/orgs/polarfire-soc/discussions/100#discussioncomment-3041503).
